### PR TITLE
Claim public or private ipfs hash

### PR DESCRIPTION
**Pull request by Holymir**
extend functionality to create public or private claims and keeping info about private ipfs address if claim is published

### CODE REVIEW

File: test/Share/IPClaimsShareTest.js [hunk 1]
```diff
@@ -248,18 +248,18 @@ contract("IPClaimsShare", (accounts) => {
...
-            let claims = await shareInstance.getClaimsCountPerRecipient(claimReceiver);
-            let _claim = await shareInstance.getClaimPerRecipient(claimReceiver, 0);
-
-            assert.strictEqual(1, recipients.toNumber(), "Recipients count does not match");
-            assert.strictEqual(claimReceiver, _claimReceiver, "Claim receiver does not match");
-            assert.strictEqual(ipfsHash, _ndaHash, "Signed NDA hash does not match");
-
-            assert.strictEqual(1, claims.toNumber(), "Claims count does not match");
-            assert.strictEqual(claim, _claim, "Claim does not match");
+            // let recipients = await shareInstance.getRecipientsCountPerClaim(claim);
```
Comment (vladichhh) [inline test/Share/IPClaimsShareTest.js]: why this snippet is commented ?

File: contracts/Factory/IPClaimsFactory.sol [hunk 2]
```diff
@@ -131,49 +131,59 @@ contract IPClaimsFactory is IIPClaimsFactory, OwnableUpgradableImplementation {
...
+    }
+
+    function createClaim(bytes32 _privateIPFSAddress, bytes32 _publicIPFSAddress, uint8 _claimType, bool _isPublic) internal {
         require(_claimType < claimTypesPricesInUSDForCreate.length, "there is no such claim type");
 
         uint256 claimRecordingFeeInTokens;
 
-        if (_isPublic == true) {
-            claimRecordingFeeInTokens = tokenExchangeOracle.convertUSDToTokens(claimTypesPricesInUSDForCreate[_claimType]);
+        if (_isPublic == false) {
```
Comment (vladichhh) [inline contracts/Factory/IPClaimsFactory.sol]: you could use 'if (_isPublic) {}' (and switch the logic from the block with those within else block)

File: contracts/Factory/IIPClaimsFactory.sol [hunk 3]
```diff
@@ -26,6 +26,7 @@ contract IIPClaimsFactory is IOwnableUpgradableImplementation {
     function setSpecificClaimTypePriceInUSDForPublish(uint8 _typeID, uint256 _typePrice) public;
     function getSpecificClaimTypePriceInUSDForPublish(uint8 _typeID) public view returns (uint256);
     function getClaimTypesCount() public view returns (uint256);
-    function createClaim(bytes32 _ipfsAddress, uint8 _claimType, bool _isPublic) public;
-    function publishClaim(address _claim) public;
+    function createPrivateClaim(bytes32 _privateIPFSAddress, uint8 _claimType, bool _isPublic) public;
```
Comment (vladichhh) [inline contracts/Factory/IIPClaimsFactory.sol]: maybe its better to be external ?

File: contracts/Factory/IIPClaimsFactory.sol [hunk 4]
```diff
@@ -26,6 +26,7 @@ contract IIPClaimsFactory is IOwnableUpgradableImplementation {
...
+    function createPublicClaim(bytes32 _publicIPFSAddress, uint8 _claimType, bool _isPublic) public;
```
Comment (vladichhh) [inline contracts/Factory/IIPClaimsFactory.sol]: maybe its better to be external ?

File: contracts/Factory/IIPClaimsFactory.sol [hunk 5]
```diff
@@ -26,6 +26,7 @@ contract IIPClaimsFactory is IOwnableUpgradableImplementation {
...
+    function publishClaim(address _claim, bytes32 _publicIPFSHash) public;
```
Comment (vladichhh) [inline contracts/Factory/IIPClaimsFactory.sol]: maybe its better to be external ?

File: contracts/Factory/IPClaimsFactory.sol [hunk 6]
```diff
@@ -131,49 +131,59 @@ contract IPClaimsFactory is IIPClaimsFactory, OwnableUpgradableImplementation {
         return claimTypesPricesInUSDForCreate.length;
     }
 
-    function createClaim(bytes32 _ipfsAddress, uint8 _claimType, bool _isPublic) public {
-        require(_ipfsAddress != bytes32(0));
+    function createPrivateClaim(bytes32 _privateIPFSAddress, uint8 _claimType, bool _isPublic) public {
+        require(_isPublic == false);
```
Comment (thcrnk) [inline contracts/Factory/IPClaimsFactory.sol]: Use only (!_isPublic)

File: contracts/Factory/IPClaimsFactory.sol [hunk 7]
```diff
@@ -131,49 +131,59 @@ contract IPClaimsFactory is IIPClaimsFactory, OwnableUpgradableImplementation {
...
+        require(_privateIPFSAddress != bytes32(0));
+        createClaim(_privateIPFSAddress, bytes32(0), _claimType, _isPublic);
+    }
+
+    function createPublicClaim(bytes32 _publicIPFSAddress, uint8 _claimType, bool _isPublic) public {
+        require(_isPublic == true);
```
Comment (thcrnk) [inline contracts/Factory/IPClaimsFactory.sol]: Use only isPublic

File: contracts/Factory/IPClaimsFactory.sol [hunk 8]
```diff
@@ -131,49 +131,59 @@ contract IPClaimsFactory is IIPClaimsFactory, OwnableUpgradableImplementation {
...
+        require(_publicIPFSAddress != bytes32(0));
+        createClaim(bytes32(0), _publicIPFSAddress, _claimType, _isPublic);
+    }
+
+    function publishClaim(address _claim, bytes32 _publicIPFSHash) public {
+        require(IPClaim(_claim).owner() == msg.sender);
+        require(IPClaim(_claim).isPublic() == false);
```
Comment (thcrnk) [inline contracts/Factory/IPClaimsFactory.sol]: use (!IPClaim(_claim).isPublic())

**Pull request merged**